An agent starting automatically for no apparent reason

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.0-beta6 | Remoted | Manager/Agent | Packages | Debian 10 |

## Description

During the development carried out in https://github.com/wazuh/wazuh-qa/issues/5199, it has been observed, after running the tests once fixed, that an agent is receiving a startup for no apparent reason.

The test flow is as follows: Register 40 agents in the master and restart them, after waiting some time, check if all of them are active, and clean the `ossec.log` of the master. Then, stops the 40 agents in parallel, waits 10 seconds and looks for 40 messages with `HC_SHUTDOWN`, and finally checks also that the 40 agents appear `Disconnected`.

The test is failing in the part of checking that all appear `Disconnected`, and if we check the logs, we see that the 40 `HC_SHUTDOWN` have been received, however, we see that there is an agent that has also received an `HC_STARTUP` later, which is the one that remains active.

Report: [big_cluster_40_agents_env.zip](https://github.com/wazuh/wazuh/files/15190237/big_cluster_40_agents_env.zip)
Agent logs: [ossec.log](https://github.com/wazuh/wazuh/files/15190504/ossec.log)

## Running the test in a local environment
### Version 4.8.0
Following the [steps](https://github.com/wazuh/wazuh/issues/23221#issuecomment-2098097651) indicated to run the test, I had to run `pip install .` in the `deps` folder to make the test work, an extra step.
``` console
cd deps/wazuh_testing
python3 setup.py install
pip install .                       <-----
```

### Test 1  (Running for the first time does not fail, the test is re-run 3 times without failing in any of the 3 cases). 🟢 

``` console
# python3 -m pytest tests/system -m big_cluster_40_agents_env -k shut --html=report.html
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.4.0
rootdir: /home/chb/workspace/wazuh-qa/tests/system, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-3.1.1
collected 192 items / 191 deselected / 1 selected                                                                                                                                                                 
tests/system/test_shutdown_message/test_shutdown_message.py .                                                                                                                                               [100%]

================================================================================================ warnings summary =================================================================================================
../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:24
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:24: PytestUnknownMarkWarning: Unknown pytest.mark.win32 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    WINDOWS = pytest.mark.win32

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:25
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:25: PytestUnknownMarkWarning: Unknown pytest.mark.linux - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    LINUX = pytest.mark.linux

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:26
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:26: PytestUnknownMarkWarning: Unknown pytest.mark.darwin - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    MACOS = pytest.mark.darwin

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:27
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:27: PytestUnknownMarkWarning: Unknown pytest.mark.sunos5 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    SOLARIS = pytest.mark.sunos5

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:29
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:29: PytestUnknownMarkWarning: Unknown pytest.mark.agent - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    AGENT = pytest.mark.agent

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
---------------------------------------------------------------------- generated html file: file:///home/chb/workspace/wazuh-qa/report.html -----------------------------------------------------------------------
============================================================================ 1 passed, 191 deselected, 5 warnings in 546.23s (0:09:06) ============================================================================
```

### Test 2  (After destroying the environment and provisioning it again, the test fails) 🔴 

``` console
cd tests/system/provisioning/big_cluster_40_agents
ansible-playbook -i inventory.yml destroy.yml
```

``` console
python3 -m pytest tests/system -m big_cluster_40_agents_env -k shut --html=report.html
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.4.0
rootdir: /home/chb/workspace/wazuh-qa/tests/system, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-3.1.1
collected 192 items / 191 deselected / 1 selected                                                                                                                                                                 

tests/system/test_shutdown_message/test_shutdown_message.py F                                                                                                                                               [100%]

==================================================================================================== FAILURES =====================================================================================================
_________________________________________________________________________________ test_shut_down_message_gracefully_stopped_agent _________________________________________________________________________________
restart_all_agents = None, stop_gracefully_all_agents = None

    def test_shut_down_message_gracefully_stopped_agent(restart_all_agents, stop_gracefully_all_agents):
        '''
            description: Checking shutdown message when socket is closed.
            wazuh_min_version: 4.6.0
            parameters:
                - restart_all_agents:
                    type: function
                    brief: Restart all the agents to manipulate them after.
                - stop_gracefully_all_agents:
                    type: function
                    brief: Stop agents gracefully
            assertions:
                - Verify that all agents status became 'Disconnected' after gracefully shutdown.

            input_description: Different use cases are found in the test module and include parameters.

            expected_output:
                - Gracefully closed, it is expected to find agents 'Disconected' in agent-manager
        '''
        time.sleep(TIMEOUT_AGENT)

        matches = re.findall(r"Disconnected", host_manager.run_command(testinfra_hosts[0],
                                                                       f'{WAZUH_PATH}/bin/agent_control -l'))


>       assert len(matches) == number_agents
E       AssertionError: assert 39 == 40
E        +  where 39 = len(['Disconnected', 'Disconnected', 'Disconnected', 'Disconnected', 'Disconnected', 'Disconnected', ...])

tests/system/test_shutdown_message/test_shutdown_message.py:99: AssertionError
================================================================================================ warnings summary =================================================================================================
../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:24
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:24: PytestUnknownMarkWarning: Unknown pytest.mark.win32 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    WINDOWS = pytest.mark.win32

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:25
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:25: PytestUnknownMarkWarning: Unknown pytest.mark.linux - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    LINUX = pytest.mark.linux

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:26
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:26: PytestUnknownMarkWarning: Unknown pytest.mark.darwin - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    MACOS = pytest.mark.darwin

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:27
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:27: PytestUnknownMarkWarning: Unknown pytest.mark.sunos5 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    SOLARIS = pytest.mark.sunos5

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:29
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:29: PytestUnknownMarkWarning: Unknown pytest.mark.agent - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    AGENT = pytest.mark.agent

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
---------------------------------------------------------------------- generated html file: file:///home/chb/workspace/wazuh-qa/report.html -----------------------------------------------------------------------
============================================================================================= short test summary info =============================================================================================
FAILED tests/system/test_shutdown_message/test_shutdown_message.py::test_shut_down_message_gracefully_stopped_agent - AssertionError: assert 39 == 40
============================================================================ 1 failed, 191 deselected, 5 warnings in 723.46s (0:12:03) ============================================================================

```

### Test 3 - (Repeating the test after failure of test 2 without re-provisioning the environment) 🟢 

``` console
# python3 -m pytest tests/system -m big_cluster_40_agents_env -k shut --html=report.html
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.4.0
rootdir: /home/chb/workspace/wazuh-qa/tests/system, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-3.1.1
collected 192 items / 191 deselected / 1 selected                                                                                                                                                                 

tests/system/test_shutdown_message/test_shutdown_message.py .                                                                                                                                               [100%]

================================================================================================ warnings summary =================================================================================================
../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:24
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:24: PytestUnknownMarkWarning: Unknown pytest.mark.win32 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    WINDOWS = pytest.mark.win32

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:25
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:25: PytestUnknownMarkWarning: Unknown pytest.mark.linux - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    LINUX = pytest.mark.linux

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:26
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:26: PytestUnknownMarkWarning: Unknown pytest.mark.darwin - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    MACOS = pytest.mark.darwin

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:27
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:27: PytestUnknownMarkWarning: Unknown pytest.mark.sunos5 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    SOLARIS = pytest.mark.sunos5

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:29
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:29: PytestUnknownMarkWarning: Unknown pytest.mark.agent - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    AGENT = pytest.mark.agent

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
---------------------------------------------------------------------- generated html file: file:///home/chb/workspace/wazuh-qa/report.html -----------------------------------------------------------------------
============================================================================ 1 passed, 191 deselected, 5 warnings in 603.80s (0:10:03) ============================================================================

```

## Conclusion
As a first observation, the test is not stable, we can say that it is a flaky test.
Interesting points to analyze:
- In the function `test_shut_down_message_gracefully_stopped_agent` there is the define `TIMEOUT_AGENT` for the sleep before the validation of the disconnected state, the value is `8` seconds, this value how was calculated? is this time enough?
- Has the same test been run in any 4.7.X version? This could be interesting, so we can rule out that it is a specific problem of version 4.8.0.
- Out of 9 times I ran the test locally, only once did it fail.

## Analyzing a failing test

#### Steps to run the test

``` console
cd tests/system/provisioning/big_cluster_40_agents

ansible-playbook -i inventory.yml playbook.yml --extra-vars='{"package_repository":"packages-dev", "repository": "pre-release", "package_version": "4.8.0", "package_revision": "1", "wazuh_qa_branch": "v4.8.0-rc1"}'

cd -

python3 -m pytest tests/system -m big_cluster_40_agents_env -k shut --html=report.html
```
#### Result

<details><summary>Result</summary>

``` console 
python3 -m pytest tests/system -m big_cluster_40_agents_env -k shut --html=report.html
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.4.0
rootdir: /home/chb/workspace/wazuh-qa/tests, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-3.1.1
collected 0 items                                                                                                                                                                                                 

------------------------------------------------------------------- generated html file: file:///home/chb/workspace/wazuh-qa/tests/report.html --------------------------------------------------------------------
============================================================================================== no tests ran in 0.01s ==============================================================================================
ERROR: file or directory not found: tests/system

root@chb-VBox:/home/chb/workspace/wazuh-qa/tests# cd ..
root@chb-VBox:/home/chb/workspace/wazuh-qa# python3 -m pytest tests/system -m big_cluster_40_agents_env -k shut --html=report.html
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.4.0
rootdir: /home/chb/workspace/wazuh-qa/tests/system, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-3.1.1
collected 192 items / 191 deselected / 1 selected                                                                                                                                                                 

tests/system/test_shutdown_message/test_shutdown_message.py F                                                                                                                                               [100%]
==================================================================================================== FAILURES =====================================================================================================
_________________________________________________________________________________ test_shut_down_message_gracefully_stopped_agent _________________________________________________________________________________

restart_all_agents = None, stop_gracefully_all_agents = None

    def test_shut_down_message_gracefully_stopped_agent(restart_all_agents, stop_gracefully_all_agents):
        '''
            description: Checking shutdown message when socket is closed.
            wazuh_min_version: 4.6.0
            parameters:
                - restart_all_agents:
                    type: function
                    brief: Restart all the agents to manipulate them after.
                - stop_gracefully_all_agents:
                    type: function
                    brief: Stop agents gracefully
            assertions:
                - Verify that all agents status became 'Disconnected' after gracefully shutdown.

            input_description: Different use cases are found in the test module and include parameters.

            expected_output:
              - Gracefully closed, it is expected to find agents 'Disconected' in agent-manager
        '''
        time.sleep(TIMEOUT_AGENT)

        matches = re.findall(r"Disconnected", host_manager.run_command(testinfra_hosts[0],
                                                                     f'{WAZUH_PATH}/bin/agent_control -l'))
>       assert len(matches) == number_agents

E       AssertionError: assert 38 == 40
E        +  where 38 = len(['Disconnected', 'Disconnected', 'Disconnected', 'Disconnected', 'Disconnected', 'Disconnected', ...])

tests/system/test_shutdown_message/test_shutdown_message.py:99: AssertionError
================================================================================================ warnings summary =================================================================================================
../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:24
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:24: PytestUnknownMarkWarning: Unknown pytest.mark.win32 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    WINDOWS = pytest.mark.win32

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:25
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:25: PytestUnknownMarkWarning: Unknown pytest.mark.linux - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    LINUX = pytest.mark.linux

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:26
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:26: PytestUnknownMarkWarning: Unknown pytest.mark.darwin - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    MACOS = pytest.mark.darwin

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:27
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:27: PytestUnknownMarkWarning: Unknown pytest.mark.sunos5 - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    SOLARIS = pytest.mark.sunos5

../../../../usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:29
  /usr/local/lib/python3.10/dist-packages/wazuh_testing/modules/__init__.py:29: PytestUnknownMarkWarning: Unknown pytest.mark.agent - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    AGENT = pytest.mark.agent

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
---------------------------------------------------------------------- generated html file: file:///home/chb/workspace/wazuh-qa/report.html -----------------------------------------------------------------------
============================================================================================= short test summary info =============================================================================================
FAILED tests/system/test_shutdown_message/test_shutdown_message.py::test_shut_down_message_gracefully_stopped_agent - AssertionError: assert 38 == 40
============================================================================ 1 failed, 191 deselected, 5 warnings in 801.35s (0:13:21) ============================================================================

```
</details>

#### Analysis

In the file `tests/system/provisioning/big_cluster_40_agents/ansible.log` I found that for this case the agent 30 fails, In the file I found that for this case agent 30 fails, but the provisioning continues without discarding the test, makes the test run fail. 

- `wazuh-agent30  failed=1`
``` console
.
.
2024-05-09 10:32:17,581 p=3075611 u=root n=ansible | wazuh-agent3               : ok=7    changed=6    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
2024-05-09 10:32:17,581 p=3075611 u=root n=ansible | wazuh-agent30              : ok=3    changed=2    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
2024-05-09 10:32:17,582 p=3075611 u=root n=ansible | wazuh-agent31              : ok=7    changed=6    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
.
```
[ansible.log](https://github.com/wazuh/wazuh/files/15267536/ansible.log)

Checking for the missing agent in the master,` wazuh-agent30` is not found.

<details><summary>agent_control command</summary>


```
docker exec b9cec3d041e0 /var/ossec/bin/agent_control -l

Wazuh agent_control. List of available agents:
   ID: 000, Name: wazuh-master (server), IP: 127.0.0.1, Active/Local
   ID: 001, Name: wazuh-agent1, IP: any, Active
   ID: 002, Name: wazuh-agent4, IP: any, Active
   ID: 003, Name: wazuh-agent10, IP: any, Active
   ID: 004, Name: wazuh-agent7, IP: any, Active
   ID: 005, Name: wazuh-agent8, IP: any, Active
   ID: 006, Name: wazuh-agent11, IP: any, Active
   ID: 007, Name: wazuh-agent2, IP: any, Active
   ID: 008, Name: wazuh-agent5, IP: any, Active
   ID: 009, Name: wazuh-agent3, IP: any, Active
   ID: 010, Name: wazuh-agent9, IP: any, Active
   ID: 011, Name: wazuh-agent6, IP: any, Active
   ID: 012, Name: wazuh-agent12, IP: any, Active
   ID: 013, Name: wazuh-agent13, IP: any, Active
   ID: 014, Name: wazuh-agent22, IP: any, Active
   ID: 015, Name: wazuh-agent19, IP: any, Active
   ID: 016, Name: wazuh-agent16, IP: any, Active
   ID: 017, Name: wazuh-agent14, IP: any, Active
   ID: 018, Name: wazuh-agent23, IP: any, Active
   ID: 019, Name: wazuh-agent17, IP: any, Active
   ID: 020, Name: wazuh-agent20, IP: any, Active
   ID: 021, Name: wazuh-agent15, IP: any, Active
   ID: 022, Name: wazuh-agent24, IP: any, Active
   ID: 023, Name: wazuh-agent21, IP: any, Active
   ID: 024, Name: wazuh-agent18, IP: any, Active
   ID: 025, Name: wazuh-agent25, IP: any, Active
   ID: 026, Name: wazuh-agent28, IP: any, Active
   ID: 027, Name: wazuh-agent31, IP: any, Active
   ID: 028, Name: wazuh-agent34, IP: any, Active
   ID: 029, Name: wazuh-agent26, IP: any, Active
   ID: 030, Name: wazuh-agent32, IP: any, Active
   ID: 031, Name: wazuh-agent35, IP: any, Active
   ID: 032, Name: wazuh-agent29, IP: any, Active
   ID: 033, Name: wazuh-agent27, IP: any, Active
   ID: 034, Name: wazuh-agent36, IP: any, Active
   ID: 035, Name: wazuh-agent33, IP: any, Active
   ID: 036, Name: wazuh-agent37, IP: any, Active
   ID: 037, Name: wazuh-agent40, IP: any, Active
   ID: 038, Name: wazuh-agent38, IP: any, Active
   ID: 039, Name: wazuh-agent39, IP: any, Active

```

</details>


### Conclusion
- An improvement that can be added to the test, is that even if you try to create 40 agents to validate if they really are 40, in the final comparison for which the test fails the 40 is hardcoded in the line `51`, `number_agents = 40`.

--- a/src/init/wazuh-client.sh
+++ b/src/init/wazuh-client.sh
@@ -29,7 +29,7 @@ LOCK_PID="${LOCK}/pid"
 # This number should be more than enough (even if it is
 # started multiple times together). It will try for up
 # to 10 attempts (or 10 seconds) to execute.
-MAX_ITERATION="10"
+MAX_ITERATION="60"
 
 MAX_KILL_TRIES=600
 

--- a/src/init/wazuh-local.sh
+++ b/src/init/wazuh-local.sh
@@ -38,7 +38,7 @@ LOCK_PID="${LOCK}/pid"
 # This number should be more than enough (even if it is
 # started multiple times together). It will try for up
 # to 10 attempts (or 10 seconds) to execute.
-MAX_ITERATION="10"
+MAX_ITERATION="60"
 
 MAX_KILL_TRIES=600
 

--- a/src/init/wazuh-server.sh
+++ b/src/init/wazuh-server.sh
@@ -42,7 +42,7 @@ LOCK_PID="${LOCK}/pid"
 # This number should be more than enough (even if it is
 # started multiple times together). It will try for up
 # to 10 attempts (or 10 seconds) to execute.
-MAX_ITERATION="10"
+MAX_ITERATION="60"
 
 MAX_KILL_TRIES=600
 

